compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -2704,7 +2704,7 @@ boolean symbolEnterAndValidateRecordVariable(BLangRecordVariable var, SymbolEnv
     }
 
     boolean validateRecordVariable(BLangRecordVariable recordVar, SymbolEnv env) {
-        BType recordType = Types.getReferredType(recordVar.getBType());
+        BType recordType = Types.getEffectiveType(Types.getReferredType(recordVar.getBType()));
```

Comment:
Why do we perform a check for the effective type without including it as a case for intersection within the switch block?

Reply:
Wouldn't that require changes to the method to be able to call it recursively?

Reply:
Yes, we need to call it recursively, but only for intersection types. Here, We perform a check on every type to determine if it is an intersection.
(We also perform this check in other parts of the code. My concern is why we treat intersections as a special case. Is this not a problem for code maintainability?)

Reply:
It isn't just intersections, we've also checked for type reference types here. Going forward, with the changes to intersection types, we will get intersection/reference types more often. And the effective type retrieval method will handle intersections, type reference types, and eventually union types in one go. 

Given that it is a check by tag, I don't think it makes much of a difference, especially because we'll have to continue to do this in some form and the union type has specific logic.